syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";


message Stock {
    string symbol = 1;
    float price = 2;
    google.protobuf.Timestamp timestamp = 3;
}


message AvgPrice {
    float price = 1;
}

service Pricing {
    rpc SavePrice(Stock) returns (Stock);
    rpc GetLatestPrice(google.protobuf.Empty) returns (Stock);
    rpc GetMvgAvg(google.protobuf.Empty) returns (AvgPrice);
}


